openmdao/core/component.py

```diff
@@ -104,6 +105,45 @@ def setup(self):
         """
         pass
 
+    def _setup_procs(self, pathname, comm, mode):
+        """
+        Distribute processors and assign pathnames.
+
+        Parameters
```

Comment:
Interesting. I just moved this to Component too on one of my branches, plus changed the docstring. Yours will get merged first for sure though.